src/main/kotlin/com/terraformation/backend/support/atlassian/request/AttachTemporaryFileHttpRequest.kt

```diff
@@ -0,0 +1,61 @@
+package com.terraformation.backend.support.atlassian.request
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties
+import com.terraformation.backend.support.atlassian.model.TemporaryAttachmentModel
+import io.ktor.client.call.body
+import io.ktor.client.request.HttpRequestBuilder
+import io.ktor.client.request.forms.ChannelProvider
+import io.ktor.client.request.forms.MultiPartFormDataContent
+import io.ktor.client.request.forms.formData
+import io.ktor.client.request.headers
+import io.ktor.client.request.setBody
+import io.ktor.client.statement.HttpResponse
+import io.ktor.http.Headers
+import io.ktor.http.HttpHeaders
+import io.ktor.http.HttpMethod
+import io.ktor.http.path
+import io.ktor.util.cio.toByteReadChannel
+import java.io.InputStream
+
+class AttachTemporaryFilesHttpRequest(
+    serviceDeskId: Int,
+    inputStream: InputStream,
+    private val filename: String,
+    private val contentType: String?,
+    private val fileSize: Long?,
```

Comment:
I had a similar situation in the S3 photo upload code where I needed an input stream but also needed to know its size and content type ahead of time. I made `SizedInputStream` to bundle those things together. Might be useful here too.